eng/Build.props

```diff
@@ -130,6 +130,14 @@
         <ProjectToBuild Condition=" '$(BuildNodeJS)' == 'true'" Include="@(NodeJsProjects)" Exclude="@(ProjectToExclude)" />
         <ProjectToExclude Condition=" '$(BuildNodeJS)' != 'true'" Include="@(NodeJsProjects)" />
 
+        <YarnMSBuildProjects Include="
```

Comment:
There is a bizarre issue here where `_IsProjectRestoreSupported` is invoked by Arcade's Build.proj and evaluates all projects without filtering against `<ExcludeFromSourceBuild>`. The only way I can think of is to filter it here but I think this approach is error prone so I'm not terribly happy with this "fix".

Reply:
Not so bizarre&hellip; Arcade's Build.proj and our Build.props don't import any Directory.Build.* files. That means `restore` operations happen for numerous projects we won't actually build.

Another workaround might be to copy something like the `$(Is{Whatever})`, `$(ExcludeFromSourceBuild)`, and maybe `$(ExcludeFromBuild)` settings from root Directory.Build.* into this file. It can't be the same because `$(MSBuildProjectName)` will always be the Arcade Build.proj file in here. But, could have similar exclusions when `$(DotNetBuildFromSource)` and perhaps `$(SkipTestBuild)` are set globally.

All that said, https://github.com/dotnet/arcade/blob/main/src/Microsoft.DotNet.Arcade.Sdk/tools/Build.proj#L124-L128 _should_ mean restoring everything doesn't matter much (beyond a few wasted CPU cycles and HTTPS downloads).

Reply:
Ah well I'm okay with this then.